internal/directives/simple_engine.go

```diff
@@ -13,6 +15,8 @@ import (
 	"github.com/akuity/kargo/internal/credentials"
 )
 
+var forbiddenStepAliasRegex = regexp.MustCompile(`^step-\d+$`)
```

Comment:
Maybe better to make this publicly accessible and have the webhook depend on it?

Reply:
I didn't want to make the webhook dependent on this package or vice versa, but since you've called me out on this, I probably _should_ move this to `api/v1alpha1`. Pretty much everything already depends on that and that is arguably a better location for this anyway.

Fix incoming.